python/pyhail/dataset.py

```diff
@@ -778,8 +1129,35 @@ def filter_variants_expr(self, condition, keep=True):
     def filter_variants_intervals(self, input, keep=True):
         """Filter variants with an .interval_list file.
 
+        **Examples**
+
+        If ``intervals.txt`` contains intervals in the interval_list format, the
+        following expression will produce a :py:class:`.VariantDataset` containg
+        only variants included by the given intervals:
+
+        >>> vds.filter_variants_intervals('data/intervals.txt')
+
+        **The interval_list Format**
+
+        Hail expects an interval file to contain either three or five fields per
+        line in the following formats:
+
+        - ``contig:start-end``
+        - ``contig  start  end`` (tab-separated)
+        - ``contig  start  end  direction  target`` (tab-separated)
+
+        In either case, Hail only uses the ``contig``, ``start``, and ``end``
+        fields.  Each variant is evaluated against each line in the interval
+        file, and any match will mark the variant to be included if
+        ``keep=True`` and excluded if ``keep=False``. *Note:* ``start`` and
+        ``end`` match positions inclusively, e.g. ``start <= position <= end``
+
         :param str input: Path to .interval_list file.
```

Comment:
Should we define file formats separately? Example: .interval_list, .columns, ...